packages/http-client/src/http-client.ts

```diff
@@ -301,18 +327,61 @@ function logRequestInformation(request: HttpRequestConfig) {
   }
 }
 
+/**
+ * Builds a [[DestinationHttpRequestConfig]] for the given destination, merges it into the given `requestConfig`
+ * and executes it (using Axios).
+ * The overload, that accepts [[HttpRequestConfigWithOrigin]] as a parameter, is deprecated and replaced the function [[executeHttpRequestWithOrigin]].
+ * @param destination - A destination or a destination name and a JWT.
+ * @param requestConfig - Any object representing an HTTP request.
+ * @param options - An [[HttpRequestOptions]] of the HTTP request for configuring e.g., CSRF token delegation. By default, the SDK will fetch the CSRF token.
+ * @returns A promise resolving to an [[HttpResponse]].
+ */
+export function executeHttpRequest<T extends HttpRequestConfig>(
+  destination: DestinationOrFetchOptions,
+  requestConfig: T,
+  options?: HttpRequestOptions
+): Promise<HttpResponse>;
+/**
+ * @deprecated This overload is replaced by the function [[executeHttpRequestWithOrigin]].
+ */
+export function executeHttpRequest<T extends HttpRequestConfigWithOrigin>(
+  destination: DestinationOrFetchOptions,
+  requestConfig: T,
+  options?: HttpRequestOptions
+): Promise<HttpResponse>;
+// eslint-disable-next-line jsdoc/require-jsdoc
+export function executeHttpRequest<
+  T extends HttpRequestConfig | HttpRequestConfigWithOrigin
+>(
+  destination: DestinationOrFetchOptions,
+  requestConfig: T,
+  options?: HttpRequestOptions
+): Promise<HttpResponse> {
+  // eslint-disable-next-line jsdoc/require-jsdoc
+  const requestConfigWithOrigin =
+    buildHttpRequestConfigWithOrigin(requestConfig);
+  return execute(executeWithAxios)(
+    destination,
+    requestConfigWithOrigin,
+    options
+  );
+}
+
 /**
  * Builds a [[DestinationHttpRequestConfig]] for the given destination, merges it into the given [[HttpRequestConfigWithOrigin]]
  * and executes it (using Axios).
  * The [[HttpRequestConfigWithOrigin]] supports defining header options and query parameter options with origins.
- * When reaching conflicts, values with high priorities are chosen.
- * The priorities are defined in the [[origins]].
+ * Equally named headers and query parameters are prioritized in the following order:
+ * 1. `custom`
+ * 2. `requestConfig`.
```

Comment:
[q] Isn't the destination missing now? I mean - not `destination`, but the fact that `requestConfig` would be overwritten by parameters created by the SDK. Will you add a link as discussed?